src/controller/State.java

```diff
@@ -0,0 +1,76 @@
+package controller;
+
+import java.util.ArrayList;
+
+public class State {
+
+    private String stringState;
+    private State parent;
+    private int gCost;
+    private int hCost;
+
+    public State(String stateStr) {
+        this.stringState = stateStr;
+
+    }
+    public String getStringState() {
+        return this.stringState;
+    }
+
+
+    public void setParent(State parentState) {
+        this.parent = parentState;
+    }
+    public State getParent() {
+        return this.parent;
+    }
+
+    public int getGcost() {
+        return this.gCost;
+    }
+
+    public int getHcost() {
+        return this.hCost;
+    }
+
+    public int getFcost() {
+        return this.gCost + this.hCost;
+    }
+
+
+    private ArrayList<Integer> getStatesCoordinates (State currState, State destState) {
+        ArrayList<Integer> coordinatesStates = new ArrayList<>();
+
+        // currStateX:
+        coordinatesStates.add(Integer.parseInt(currState.getStringState().substring(14).split(",")[0]
```

Comment:
Feels super (duper) specific to sit in the generic class of "State"

Reply:
This is the only relevant place I can think of. Not sure if this comment is regarding the implementation or the location of this method.

Reply:
Location
Let's think about it together tonight